core/mempool/src/lib.rs

```diff
@@ -66,7 +69,17 @@ impl Pool {
     }
 
     pub fn is_empty(&self) -> bool {
-        self.transactions.read().expect(POISONED_LOCK_ERR).is_empty() && self.receipts.read().expect(POISONED_LOCK_ERR).is_empty()
+        self.transactions.read().expect(POISONED_LOCK_ERR).is_empty() 
+        && self.receipts.read().expect(POISONED_LOCK_ERR).is_empty()
+    }
+
+    pub fn len(&self) -> usize {
```

Comment:
Why each receipt block increments length by 1 and not by the size of the receipt block?

Reply:
It's debatable. I view `ReceiptBlock` as one item. We could change to count the number of receipts if that's more preferable.